### PR TITLE
coperl4 bug fix

### PR DESCRIPTION
**Pull request by mjhossen**
## Description
This PR includes the following changes:
1.  replace the keyword 'err' by 'err_sla' in the python script
2. fixed datetime inconsistency. One needs to pass the datetime as an argument in the format: %Y-%m-%dT%H:%M:%SZ
### Issues addressed
 #955

### CODE REVIEW

**Comment by liuxiao37k** [comment]
@mjhossen I ran your branch on Orion. Your `copernicus_l4adt2ioda.py` failed the `iodaconv_marine_coding_norms` ctest. Is it the same reason why ci failed (I don't have the permission to see the log on AWS)? Did you run the coding_norm ctests locally? You have to pass all ctests to make ci happy.
```
1/1 Test #7: iodaconv_marine_coding_norms .....***Failed    0.55 sec
```
In particular,
```
7: ./copernicus_l4adt2ioda.py:76:1: E302 expected 2 blank lines, found 1
7: ./copernicus_l4adt2ioda.py:79:22: E225 missing whitespace around operator
7: ./copernicus_l4adt2ioda.py:100:17: E225 missing whitespace around operator
7: ./copernicus_l4adt2ioda.py:101:18: E225 missing whitespace around operator
7: ./copernicus_l4adt2ioda.py:102:28: E225 missing whitespace around operator
```

**Comment by liuxiao37k** [comment]
@mjhossen Also, you don't need to create a [remote branch in ioda-bundle](https://github.com/JCSDA-internal/ioda-bundle/tree/feature/coperl4_bugfixed) to merge this PR. You only need to do that in your local branch to build ioda-converters and run ctests. Plus, in the `CMakelist.txt` that your changed, it is pointing to the `feature/filter_emcice` branch instead of `feature/coperl4_bugfixed`.
https://github.com/JCSDA-internal/ioda-bundle/blob/797423171a1cdce4846896d2eb191724f69ff062/CMakeLists.txt#L61

**Comment by mjhossen** [comment]
> @mjhossen Also, you don't need to create a [remote branch in ioda-bundle](https://github.com/JCSDA-internal/ioda-bundle/tree/feature/coperl4_bugfixed) to merge this PR. You only need to do that in your local branch to build ioda-converters and run ctests. Plus, in the `CMakelist.txt` that your changed, it is pointing to the `feature/filter_emcice` branch instead of `feature/coperl4_bugfixed`. https://github.com/JCSDA-internal/ioda-bundle/blob/797423171a1cdce4846896d2eb191724f69ff062/CMakeLists.txt#L61

I again run it and it is passed. Here is the output:
(soca_python-3.9) (base) Orion-login-1[11] jhossen$ ctest -I 1,18
Test project /work/noaa/ng-godas/jhossen/s2s_sprint/ioda-bundle/build/iodaconv/test
      Start  1: test_iodaconv_gds2_sst_l2p
 1/18 Test  #1: test_iodaconv_gds2_sst_l2p .......   Passed    7.74 sec
      Start  2: test_iodaconv_gds2_sst_l3u
 2/18 Test  #2: test_iodaconv_gds2_sst_l3u .......   Passed    0.58 sec
      Start  3: test_iodaconv_smap_sss
 3/18 Test  #3: test_iodaconv_smap_sss ...........   Passed    0.56 sec
      Start  4: test_iodaconv_rads_adt
 4/18 Test  #4: test_iodaconv_rads_adt ...........   Passed    0.51 sec
      Start  5: test_iodaconv_godae_prof
 5/18 Test  #5: test_iodaconv_godae_prof .........   Passed    0.65 sec
      Start  6: test_iodaconv_godae_ship
 6/18 Test  #6: test_iodaconv_godae_ship .........   Passed    0.57 sec
      Start  7: test_iodaconv_godae_trak
 7/18 Test  #7: test_iodaconv_godae_trak .........   Passed    0.56 sec
      Start  8: test_iodaconv_hgodas_adt
 8/18 Test  #8: test_iodaconv_hgodas_adt .........   Passed    0.50 sec
      Start  9: test_iodaconv_hgodas_insitu
 9/18 Test  #9: test_iodaconv_hgodas_insitu ......   Passed    0.53 sec
      Start 10: test_iodaconv_hgodas_sst
10/18 Test #10: test_iodaconv_hgodas_sst .........   Passed    0.51 sec
      Start 11: test_iodaconv_glider
11/18 Test #11: test_iodaconv_glider .............   Passed    1.07 sec
      Start 12: test_iodaconv_hfradar
12/18 Test #12: test_iodaconv_hfradar ............   Passed    0.57 sec
      Start 13: test_iodaconv_argoclim
13/18 Test #13: test_iodaconv_argoclim ...........   Passed    0.50 sec
      Start 14: test_iodaconv_cryosat2
14/18 Test #14: test_iodaconv_cryosat2 ...........   Passed    0.94 sec
      Start 15: test_iodaconv_emc_ice
15/18 Test #15: test_iodaconv_emc_ice ............***Failed    0.38 sec
      Start 16: test_iodaconv_smos_sss
16/18 Test #16: test_iodaconv_smos_sss ...........   Passed    2.44 sec
      Start 17: test_iodaconv_copernicus_l4adt
17/18 Test #17: test_iodaconv_copernicus_l4adt ...   Passed    0.66 sec
      Start 18: test_iodaconv_copernicus_l3swh
18/18 Test #18: test_iodaconv_copernicus_l3swh ...   Passed    0.61 sec

I think you need to provide pythonpath, such as:
export IODA=/work/noaa/ng-godas/jhossen/s2s_sprint/ioda-bundle/build/bin
source /work/noaa/ng-godas/jhossen/s2s_sprint/soca-science/configs/machine/machine.orion.intel
pyioda_path=${IODA}/../../../build/lib/python3.9/pyioda
export PYTHONPATH=$PYTHONPATH:$pyioda_path

**Comment by liuxiao37k** [comment]
@mjhossen As I have already mentioned above, your local ioda-bundle CMakelist.txt on Orion points to the `feature/filter-emcice` branch where you made changes to the ice converter only. Hence, you passed the adt converter ctest, and **failed** the ice converter ctest, as you have already shown above (Test 15/18). This is a PR for your adt converter change, based on the description of this PR.
Also, again, you need to make all ctests (not just what's under /marine) to make `ci` happy. Did you run those `_coding_norm` ctests locally? Your converter can convert obs file correctly, but you need to add those whitespaces and empty lines to fulfill `pycodestyle`.
```
Start 15: test_iodaconv_emc_ice
15/18 Test 15: test_iodaconv_emc_ice ............***Failed 0.38 sec
```
Orion: 
/work/noaa/ng-godas/jhossen/s2s_sprint/ioda-bundle/CMakeLists.txt
Line 61

Have a good weekend!

**Comment by liuxiao37k** [comment]
> @mjhossen I ran your branch on Orion. Your `copernicus_l4adt2ioda.py` failed the `iodaconv_marine_coding_norms` ctest. Is it the same reason why ci failed (I don't have the permission to see the log on AWS)? Did you run the `_coding_norm` ctests locally? You have to pass all ctests to make ci happy.
> 
> ```
> 1/1 Test #7: iodaconv_marine_coding_norms .....***Failed    0.55 sec
> ```
> 
> Specifically,
> 
> ```
> 7: ./copernicus_l4adt2ioda.py:76:1: E302 expected 2 blank lines, found 1
> 7: ./copernicus_l4adt2ioda.py:79:22: E225 missing whitespace around operator
> 7: ./copernicus_l4adt2ioda.py:100:17: E225 missing whitespace around operator
> 7: ./copernicus_l4adt2ioda.py:101:18: E225 missing whitespace around operator
> 7: ./copernicus_l4adt2ioda.py:102:28: E225 missing whitespace around operator
> ```

@ShastriPaturi @guillaumevernieres FYI, the above error messages reported by my local ctests are consistent with what I have seen from the logs on AWS from the `travis-ci` test. Hope the information can assist with your review.

**Comment by mjhossen** [comment]
> > @mjhossen I ran your branch on Orion. Your `copernicus_l4adt2ioda.py` failed the `iodaconv_marine_coding_norms` ctest. Is it the same reason why ci failed (I don't have the permission to see the log on AWS)? Did you run the `_coding_norm` ctests locally? You have to pass all ctests to make ci happy.
> > ```
> > 1/1 Test #7: iodaconv_marine_coding_norms .....***Failed    0.55 sec
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Specifically,
> > ```
> > 7: ./copernicus_l4adt2ioda.py:76:1: E302 expected 2 blank lines, found 1
> > 7: ./copernicus_l4adt2ioda.py:79:22: E225 missing whitespace around operator
> > 7: ./copernicus_l4adt2ioda.py:100:17: E225 missing whitespace around operator
> > 7: ./copernicus_l4adt2ioda.py:101:18: E225 missing whitespace around operator
> > 7: ./copernicus_l4adt2ioda.py:102:28: E225 missing whitespace around operator
> > ```
> 
> @ShastriPaturi @guillaumevernieres FYI, the above error messages reported by my local ctests are consistent with what I have seen from the logs on AWS from the `travis-ci` test. Hope the information can assist with your review.

@liuxiao37k though I fixed mine, ctest will still fail for other code. For example:
7: checking ./emc_ice2ioda.py
7: ./emc_ice2ioda.py:49:32: E201 whitespace after '('
7: ./emc_ice2ioda.py:49:64: E225 missing whitespace around operator
7: ./emc_ice2ioda.py:49:70: E202 whitespace before ')'
7: ./emc_ice2ioda.py:51:32: E201 whitespace after '('
7: ./emc_ice2ioda.py:51:64: E225 missing whitespace around operator
7: ./emc_ice2ioda.py:51:71: E202 whitespace before ')'
7: ./emc_ice2ioda.py:77:1: E305 expected 2 blank lines after class or function definition, found 1

**Review by liuxiao37k — COMMENTED** [review]
This branch passed all ctests on Orion. One question before approving: is there a particular reason to replace the existing input data file (dt_global_twosat_phy_l4_20150101_vDT2018.nc)? Has the format of input data been changed?

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
Thanks @mjhossen . Looks fine except for the ``IODA_CONV_PATH`` that needs to be reverted.

**Comment by ShastriPaturi** [comment]
> This branch passed all ctests on Orion. One question before approving: is there a particular reason to replace the existing input data file (dt_global_twosat_phy_l4_20150101_vDT2018.nc)? Has the format of input data been changed?

@liuxiao37k I have a similar question... The 20150101 files have been replaced by 20190101. My tests pass on orion as well.
I am ready to approve as well.

File: src/marine/copernicus_l4adt2ioda.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import os
 from pathlib import Path
 
-IODA_CONV_PATH = Path(__file__).parent/"@SCRIPT_LIB_PATH@"
+IODA_CONV_PATH = Path(__file__).parent/"../lib/pyiodaconv"
```
[the remark this answers is not in the document]
Comment (mjhossen) [reply src/marine/copernicus_l4adt2ioda.py]: done

**Comment by mjhossen** [comment]
> This branch passed all ctests on Orion. One question before approving: is there a particular reason to replace the existing input data file (dt_global_twosat_phy_l4_20150101_vDT2018.nc)? Has the format of input data been changed?

@liuxiao37k some keywords are changed in the recent data format. New keywords are associated with the current datafile, e.g 2019-present.

File: src/marine/copernicus_l4adt2ioda.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liuxiao37k) [reply src/marine/copernicus_l4adt2ioda.py]: I think @guillaumevernieres meant to ask you to only revert line 18, not the line below
```
if not IODA_CONV_PATH.is_dir():
IODA_CONV_PATH = Path(__file__).parent/'..'/'lib-python'
```
Can you confirm @guillaumevernieres ?

[the remark this answers is not in the document]
Comment (mjhossen) [reply src/marine/copernicus_l4adt2ioda.py]: corrected!

**Review by liuxiao37k — APPROVED** [review]

**Review by ShastriPaturi — APPROVED** [review]

**Review by guillaumevernieres — APPROVED** [review]

**Review by ShastriPaturi — APPROVED** [review]

**Comment by liuxiao37k** [comment]
@mjhossen reminder to add JCSDA review

**Pull request merged**